dhcp-protocol/src/main/java/org/anarres/dhcp/v6/options/IaAddressOption.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package org.anarres.dhcp.v6.options;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import javax.annotation.Nonnull;
+import org.anarres.dhcp.v6.io.Dhcp6MessageEncoder;
+import org.apache.directory.server.dhcp.DhcpException;
+
+/**
+ * https://tools.ietf.org/html/rfc3315#section-22.6
+ *
+ * @author marosmars
+ */
+public class IaAddressOption extends SuboptionOption {
+
+    private static final short TAG = 5;
+    private static final int HEADER_SIZE = 24;
+
+    @Override
+    public short getTag() {
+        return TAG;
+    }
+
+    @Nonnull
+    public InetAddress getIp() {
+        ByteBuffer buf = ByteBuffer.wrap(getData());
+        final byte[] dst = new byte[16];
+        buf.get(dst, 0, 16);
+
+        try {
+            return Inet6Address.getByAddress(dst);
+        } catch (UnknownHostException e) {
+            // This can happen only if the IP byte array is of illegal length
+            throw new IllegalStateException("Illegal IP address", e);
+        }
+    }
+
+    public void setIp(@Nonnull final InetAddress address) {
+        Preconditions.checkArgument(address instanceof Inet6Address);
+        ByteBuffer buf = ByteBuffer.wrap(getData());
+        buf.put(address.getAddress());
```

Comment:
This rather assumes data non-null and of the correct size. I think `setData(address.getAddress())` is safer?